pkg/management/api/grpc/application_converter.go

```diff
@@ -0,0 +1,220 @@
+	return &proj_model.Application{
+		ObjectRoot: models.ObjectRoot{
+			ID: app.ProjectId,
+		},
+		AppID: app.Id,
+		Name:  app.Name,
+	}
+}
+
+func oidcConfigUpdateToModel(app *OIDCConfigUpdate) *proj_model.OIDCConfig {
+	return &proj_model.OIDCConfig{
+		ObjectRoot: models.ObjectRoot{
+			ID: app.ProjectId,
+		},
+		AppID:                  app.ApplicationId,
+		RedirectUris:           app.RedirectUris,
+		ResponseTypes:          oidcResponseTypesToModel(app.ResponseTypes),
+		GrantTypes:             oidcGrantTypesToModel(app.GrantTypes),
+		ApplicationType:        oidcApplicationTypeToModel(app.ApplicationType),
+		AuthMethodType:         oidcAuthMethodTypeToModel(app.AuthMethodType),
+		PostLogoutRedirectUris: app.PostLogoutRedirectUris,
+	}
+}
+
+func appStateFromModel(state proj_model.AppState) AppState {
+	switch state {
+	case proj_model.APPSTATE_ACTIVE:
+		return AppState_APPSTATE_ACTIVE
+	case proj_model.APPSTATE_INACTIVE:
+		return AppState_APPSTATE_INACTIVE
+	default:
+		return AppState_APPSTATE_UNSPECIFIED
+	}
+}
+
+func oidcResponseTypesToModel(responseTypes []OIDCResponseType) []proj_model.OIDCResponseType {
+	oidcResponseTypes := make([]proj_model.OIDCResponseType, len(responseTypes))
+
+	for i, responseType := range responseTypes {
+		switch responseType {
+		case OIDCResponseType_OIDCRESPONSETYPE_CODE:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_CODE
+		case OIDCResponseType_OIDCRESPONSETYPE_ID_TOKEN:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_ID_TOKEN
+		case OIDCResponseType_OIDCRESPONSETYPE_TOKEN_ID_TOKEN:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_TOKEN_ID_TOKEN
+		}
+	}
+
+	return oidcResponseTypes
+}
+
+func oidcResponseTypesFromModel(responseTypes []proj_model.OIDCResponseType) []OIDCResponseType {
+	oidcResponseTypes := make([]OIDCResponseType, len(responseTypes))
+
+	for i, responseType := range responseTypes {
+		switch responseType {
+		case proj_model.OIDCRESPONSETYPE_CODE:
+			oidcResponseTypes[i] = OIDCResponseType_OIDCRESPONSETYPE_CODE
+		case proj_model.OIDCRESPONSETYPE_ID_TOKEN:
+			oidcResponseTypes[i] = OIDCResponseType_OIDCRESPONSETYPE_ID_TOKEN
+		case proj_model.OIDCRESPONSETYPE_TOKEN_ID_TOKEN:
+			oidcResponseTypes[i] = OIDCResponseType_OIDCRESPONSETYPE_TOKEN_ID_TOKEN
+		default:
+			oidcResponseTypes[i] = OIDCResponseType(99)
+		}
+	}
+
+	return oidcResponseTypes
+}
+
+func oidcGrantTypesToModel(grantTypes []OIDCGrantType) []proj_model.OIDCGrantType {
+	oidcGrantTypes := make([]proj_model.OIDCGrantType, len(grantTypes))
+
+	for i, grantType := range grantTypes {
+		switch grantType {
+		case OIDCGrantType_OIDCGRANTTYPE_AUTHORIZATION_CODE:
+			oidcGrantTypes[i] = proj_model.OIDCGRANTTYPE_AUTHORIZATION_CODE
+		case OIDCGrantType_OIDCGRANTTYPE_IMPLICIT:
+			oidcGrantTypes[i] = proj_model.OIDCGRANTTYPE_IMPLICIT
+		case OIDCGrantType_OIDCGRANTTYPE_REFRESH_TOKEN:
+			oidcGrantTypes[i] = proj_model.OIDCGRANTTYPE_REFRESH_TOKEN
+		}
```

Comment:
add default

Reply:
i think that doesnt make sense, it should be one of these 3 types if something else happens i dont want to do something